core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -1952,6 +1958,8 @@ private StageExecution(
             this.taskSource = requireNonNull(taskSource, "taskSource is null");
             this.sinkPartitioningScheme = requireNonNull(sinkPartitioningScheme, "sinkPartitioningScheme is null");
             this.exchange = requireNonNull(exchange, "exchange is null");
+            requireNonNull(sourceStageExecutions, "sourceStageExecutions is null");
```

Comment:
```suggestion
this.sourceStageExecutions = ImmutableMap.copyOf(requireNonNull(sourceStageExecutions, "sourceStageExecutions is null"));
```

clearly optional nit.